src/ucp/core/ucp_mm.c

```diff
@@ -428,7 +437,7 @@ ucp_memh_get_slow(ucp_context_h context, void *address, size_t length,
     status         = ucp_memh_register(context, memh,
                                        ~memh->md_map & reg_md_map,
                                        reg_address, reg_length,
-                                       uct_flags);
+                                       uct_flags | UCT_MD_MEM_FLAG_HIDE_ERRORS);
```

Comment:
Can you explain what problem does it solve?
AFAIK ucp_memh_get() should hide errors only if uct_flags contains HIDE_ERRORS flag and not always.
At least that was ucp_datatype_iter_mem_reg_internal was doing.

Reply:
This pull request fixes failure while calling `ucp_mem_map` with `UCP_MEM_MAP_ALLOCATE` parameter on some systems. E.g. simple command `ucx_info -m 4096 -u t` fails on Ubuntu 20.04 because of failing of memory registration on KNEM.
Previously (before https://github.com/openucx/ucx/pull/7811) we always call `ucp_mem_rereg_mds` with `UCT_MD_MEM_FLAG_HIDE_ERRORS`. That is why there was no error.

Reply:
Adding this flag here is too broad, new proto flow shouldn't hide errors.
Need to add it in `ucp_mm_get_alloc_md_map()`, `ucp_mem_map()`, `ucp_rndv_frag_malloc_mpools()` and `ucp_mpool_malloc()`.
I would remove `alloc_md_map`/`_initialized`, make `ucp_mm_get_alloc_md_map()` code unconditional part of context initialization and `&` it's outcome with `reg_md_map[UCS_MEMORY_TYPE_HOST]`
@yosefe WDYT?